src/nunavut/lang/cpp/support/variable_length_array.hpp

```diff
@@ -9,25 +9,56 @@
 #define NUNAVUT_SUPPORT_VARIABLE_LENGTH_ARRAY_HPP_INCLUDED
 
 #include <cstddef>
+#include <cstdlib>
 #include <cstring>
 #include <type_traits>
 #include <memory>
 #include <utility>
+#include <initializer_list>
+#if __cpp_exceptions
+#include <stdexcept>
+#endif
 
 namespace nunavut
 {
 namespace support
 {
+
+/**
+ * Default allocator using malloc and free.
+ */
+template <typename T>
+class MallocAllocator
```

Comment:
We're driving towards no malloc or free in the default generated c++ but we're not quite there yet. This is an improvement from relying on std::vector, however, since that's a pretty deep dependency into the STL including a use of the default `Allocator`.

Besides malloc and free, we currently use a lot of static/compile-time support from the standard library, `std::array`, and (if generating c++17) `std::variant`. Neither variant nor array allocate memory internally.